lib/axiom/fs/stream/writable_file_stream_buffer.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AxiomError from 'axiom/core/error';
+import AxiomEvent from 'axiom/core/event';
+
+import DataType from 'axiom/fs/data_type';
+import SeekWhence from 'axiom/fs/seek_whence';
+import StreamBuffer from 'axiom/fs/stream/stream_buffer';
+import WritableMemoryStream from 'axiom/fs/stream/writable_memory_stream';
+
+/** @typedef FileSystem$$module$axiom$fs$base$file_system */
+var FileSystem;
+
+/** @typedef OpenMode$$module$axiom$fs$open_mode */
+var OpenMode;
+
+/** @typedef Path$$module$axiom$fs$path */
+var Path;
+
+/** @typedef WritableStream$$module$axiom$fs$stream$writable_stream */
+var WritableStream;
+
+/**
+ * @constructor @extends {StreamBuffer}
+ *
+ * @param {!FileSystem} fileSystem
+ * @param {Path} filePath
+ * @param {?DataType} dataType
+ * @param {!OpenMode} openMode
+ */
+export var WritableFileStreamBuffer = function(
+    fileSystem, filePath, dataType, openMode) {
+  StreamBuffer.call(this);
+
+  /** @const @type {!WritableStream} */
+  this.writableStream = new WritableMemoryStream(this);
+
+  this.writableStream.onFinish.listenOnce(function() {
+    // This will flush unsaved data to the file and fire onFlush.
+    StreamBuffer.prototype.flush.call(this);
+  }.bind(this));
+  
+  dataType = dataType || DataType.UTF8String;
+
+  fileSystem.createOpenContext(filePath, openMode)
```

Comment:
Same remark as for ReadableFileStreamBuffer, I would move this to a separate method.